recipe/meta.yaml

```diff
@@ -11,29 +11,34 @@ source:
   sha256: {{ sha256 }}
 
 build:
-  noarch: python
   number: 0
   script: "{{ PYTHON }} -m pip install . --no-deps -vv"
   entry_points:
     - black = black:main
 
 requirements:
   host:
-    - python >=3.6
+    - python
     - pip
+    - setuptools >=45.0
+    - setuptools_scm[toml] >=6.3.1
+    - wheel
   run:
-    - python >=3.6
-    - click >=6.5
-    - attrs >=18.1.0
-    - appdirs
-    - toml >=0.9.4
+    - python
+    - click >=8.0.0
+    - platformdirs >=2
+    - tomli >=1.1.0 # [py<311]
+    - typed-ast >=1.4.2 # [py<38]
+    - pathspec >=0.9.0
+    - typing_extensions >=3.10.0.0 # [py<310]
+    - mypy_extensions >=0.4.3
 
 test:
   commands:
     - black --help
 
```

Comment:
It might be interesting to add extra outputs for blackd and such.
There was some work done this way on cf: https://github.com/conda-forge/black-feedstock/blob/main/recipe/meta.yaml

Please also check the build log to verify that pip does not download dependencies. If it does, we might want to apply the same that cf did:
https://github.com/conda-forge/black-feedstock/blob/main/recipe/build_base.sh